datashare-index/src/main/java/org/icij/datashare/text/indexing/elasticsearch/ElasticsearchSearcher.java

```diff
@@ -154,19 +159,41 @@ public Indexer.Searcher set(JsonNode jsonQuery) {
         return this;
     }
 
+    @Override
+    public Indexer.Searcher setFromTemplate(String jsonQueryTemplate, String query) {
+        boolean phraseMatch = phraseMatchTemplate.matcher(jsonQueryTemplate).matches();
+
+        int fuzziness = 0;
+        Matcher fuzzinessMatcher = fuzzinessTemplate.matcher(jsonQueryTemplate);
+        if (fuzzinessMatcher.matches()) {
+            fuzziness = Integer.parseInt(fuzzinessMatcher.group(1));
+        }
+
+        String queryString = buildQueryString(query, fuzziness, phraseMatch, "\\\\\"");
+        jsonQueryTemplate = queryTemplate.matcher(jsonQueryTemplate).replaceAll(queryString);
+        final String queryBody = jsonQueryTemplate;
+
+        this.boolQueryBuilder.must(m -> m.withJson(new StringReader(queryBody)));
+        return this;
+    }
+
```

Comment:
I'm not sure to understand why we need those regexps?

As it is string manipulation, could we do that in the client?
Or could we use the BatchSearch object fuzziness/phrase match?

Reply:
I thought that the queryBody could have also contained information about the fuzziness and phrase match but the information is already in the `BatchSearch` object indeed, the implementation would be easier and more straightforward with it.

The client can't do the whole string manipulation because it doesn't know the query, for example in the case where the `<query>` contains multiple words and the phrase_match is off it would have to build something like `term01~1 term02~1 ...`

Reply:
Anyway I'll do the change, the front will just send a `<query>` to replace and the back will use the BatchSearch object to get the fuzziness and the phrase_match and the regexp won't be needed anymore